### DISCUSSION TITLE
Why would a mask affect the sibling's filter?

### DISCUSSION THREAD

**Discussion by a-c-sreedhar-reddy** [pixijs/pixijs#7656] [Q&A] [answered]
```
Container 
    Sprite - filter
    Mask
```

Hi , in the above scenario the mask affects the filter.  What could be the reason for that?

eg:
 https://jsfiddle.net/acsreedharreddy/fgy8hoqv/ This does not work as expected.
 

- In the above fiddle if we comment out `line 143: containeer.mask = mask;` The flip works fine. [fiddle](https://jsfiddle.net/acsreedharreddy/fgy8hoqv/11/)

 

- But if we add the mask to the container, then the filter behaves differently even though `uMatrix`  uniform of the filter is same for both the cases.

**Comment by ivanpopelyshev** [comment, marked as answer]
you apply uMatrix to normalized coords, right? and they are changed in case of mask because mask changes the filter bounds. 
Work with pixel coords and you wont have that problem.

changed bounds => changed texture size and projectionMatrix

**Reply by ivanpopelyshev** [reply]
Mask is also a filter, above you , and not in the same stack , so it can affect all the sizes :)

**Reply by ivanpopelyshev** [reply]
That's another point why I always work with pixels in pixi-projection and dont try relative sizes.

**Reply by ivanpopelyshev** [reply]
If you want a workaround, use SpriteMaskFilter instead of mask

**Reply by a-c-sreedhar-reddy** [reply]
I tried SpriteMaskFilter but did not work.
 https://jsfiddle.net/acsreedharreddy/fgy8hoqv/63/

**Reply by ivanpopelyshev** [reply]
I meant on sprite itself. if its on other element, "above" it, then its still the same problem - it has different width/heights and your backdrop is different.

**Reply by ivanpopelyshev** [reply]
you can try `container.filterArea = app.screen` , that way backdrop size will be the same as before , screen

**Reply by ivanpopelyshev** [reply]
Look in my latest comment about `container.filterArea = renderer.screen` for fast workaround if you dont have time to fix that

**Reply by a-c-sreedhar-reddy** [reply]
I tried it https://jsfiddle.net/acsreedharreddy/fgy8hoqv/290/ Did not have any effect.
```js
sprite.filterArea = new PIXI.Rectangle(0,0,400,400)
containeer.filterArea = new PIXI.Rectangle(0,0,400,400)
```

But applying the uMatrix to the pixel coordinate system worked.
```js
attribute vec2 aVertexPosition;
attribute vec2 aTextureCoord;

uniform mat3 projectionMatrix;
uniform mat4 uMatrix;

varying vec2 vTextureCoord;
void main(void)
{
    vec4 pos = uMatrix * vec4(aVertexPosition, 0.0, 1.0);
    gl_Position = vec4(projectionMatrix * vec3(pos.xyz), pos.w);
    vTextureCoord = aTextureCoord;
}
```
https://jsfiddle.net/acsreedharreddy/fgy8hoqv/292/

What is the third element in the vector multiplied to the projectionMatrix?
eg:` projectionMatrix * (x, y, third?)`

**Reply by ivanpopelyshev** [reply]
third? its always 1 , its usual 3x3 affine matrix.

 you should pass "pos.z" as it is

`vec4((projectionMatrix * vec3(pos.xy, 1.0)).xy , pos.zw);`
